smithy-linters/src/main/java/software/amazon/smithy/linters/MissingSensitiveTraitValidator.java

```diff
@@ -0,0 +1,233 @@
+            Set<String> wordsInit = new HashSet<>(DEFAULT_SENSITIVE_WORDS);
+            phrasesInit.addAll(config.getPhrases()
+                    .stream()
+                    .map(phrase -> phrase.toLowerCase(Locale.US))
+                    .collect(Collectors.toSet()));
+            wordsInit.addAll(config.getWords()
+                    .stream()
+                    .map(word -> word.toLowerCase(Locale.US))
+                    .collect(Collectors.toSet()));
+            sensitivePhrases = Collections.unmodifiableSet(phrasesInit);
+            sensitiveWords = Collections.unmodifiableSet(wordsInit);
+        } else {
+            if (config.getPhrases().isEmpty() && config.getWords().isEmpty()) {
+                //This configuration combination makes the validator a no-op.
+                throw new IllegalArgumentException("Cannot set 'excludeDefaults' to true and leave "
+                                                 + "both 'phrases' and 'words' unspecified.");
+            }
+            sensitivePhrases = Collections.unmodifiableSet(new HashSet<>(
+                    config.getPhrases()
+                            .stream()
+                            .map(phrase -> phrase.toLowerCase(Locale.US))
+                            .collect(Collectors.toSet()))
+            );
+            sensitiveWords = Collections.unmodifiableSet(new HashSet<>(
+                    config.getWords()
+                            .stream()
+                            .map(word -> word.toLowerCase(Locale.US))
+                            .collect(Collectors.toSet())));
+        }
+    }
+
+    /**
+     * Finds shapes without the sensitive trait that possibly contain sensitive data,
+     * based on the shape/member name and the list of key words and phrases.
+     *
+     * @param model Model to validate.
+     * @return list of violation events
+     */
+    @Override
+    public List<ValidationEvent> validate(Model model) {
+        List<ValidationEvent> validationEvents = new ArrayList<>();
+        validationEvents.addAll(scanShapeNames(model));
+        validationEvents.addAll(scanMemberNames(model));
+        return validationEvents;
+    }
+
+    private List<ValidationEvent> scanShapeNames(Model model) {
+        return model.shapes()
+                .filter(shape -> !shape.isMemberShape()
+                        && !shape.isOperationShape()
+                        && !shape.isServiceShape()
+                        && !shape.isResourceShape())
+                .filter(shape -> !shape.hasTrait(SensitiveTrait.ID))
+                .map(shape -> detectSensitiveWord(shape.toShapeId().getName(), shape).map(Optional::of)
+                        .orElse(detectSensitivePhrase(shape.toShapeId().getName(), shape)))
+                .flatMap(o -> o.map(Stream::of).orElseGet(Stream::empty))
+//                Once we finally upgrade from Java 8, above line can be simplified to
```

Comment:
I also think this would be much more readable with just normal for loops... which also removes the need for this comment :)

Reply:
Yeah, that's definitely true actually